From: Omar Plummer <omarplummer@imergetechnologies.com>
Date: Thu, 06 Aug 2015 22:24:27 +0000
Subject: Fix TagLib version check.
X-Git-Url: http://quickgit.kde.org/?p=amarok.git&a=commitdiff&h=fec8817c530ce42b6485fdabff6051784b8d9904
---
Fix TagLib version check.

The version check compared version strings lexicographically, so "1.7"
(the min version) compared as greater than "1.10.0".

BUG: 351013
REVIEW: 124639
---


--- a/cmake/modules/FindTaglib.cmake
+++ b/cmake/modules/FindTaglib.cmake
@@ -29,10 +29,10 @@
 
   exec_program(${TAGLIBCONFIG_EXECUTABLE} ARGS --version RETURN_VALUE _return_VALUE OUTPUT_VARIABLE TAGLIB_VERSION)
 
-  if(TAGLIB_VERSION STRLESS "${TAGLIB_MIN_VERSION}")
+  if("${TAGLIB_MIN_VERSION}" VERSION_GREATER TAGLIB_VERSION)
      message(STATUS "TagLib version too old: version searched :${TAGLIB_MIN_VERSION}, found ${TAGLIB_VERSION}")
      set(TAGLIB_FOUND FALSE)
-  else(TAGLIB_VERSION STRLESS "${TAGLIB_MIN_VERSION}")
+  else("${TAGLIB_MIN_VERSION}" VERSION_GREATER TAGLIB_VERSION)
 
      exec_program(${TAGLIBCONFIG_EXECUTABLE} ARGS --libs RETURN_VALUE _return_VALUE OUTPUT_VARIABLE TAGLIB_LIBRARIES)
 
@@ -42,7 +42,7 @@
         set(TAGLIB_FOUND TRUE)
      endif(TAGLIB_LIBRARIES AND TAGLIB_CFLAGS)
      string(REGEX REPLACE " *-I" ";" TAGLIB_INCLUDES "${TAGLIB_CFLAGS}")
-  endif(TAGLIB_VERSION STRLESS "${TAGLIB_MIN_VERSION}") 
+  endif("${TAGLIB_MIN_VERSION}" VERSION_GREATER TAGLIB_VERSION)
   mark_as_advanced(TAGLIB_CFLAGS TAGLIB_LIBRARIES TAGLIB_INCLUDES)
 
 else(TAGLIBCONFIG_EXECUTABLE)

